test/e2e/datafederation_pe_test.go

```diff
@@ -54,175 +64,63 @@ var _ = Describe("UserLogin", Label("datafederation"), func() {
 		})
 	})
 
-	DescribeTable("Namespaced operators working only with its own namespace with different configuration",
-		func(test *model.TestDataProvider, pe []privateEndpoint) {
-			testData = test
-			actions.ProjectCreationFlow(test)
-			dataFederationFlow(test, providerAction, pe)
-		},
-		Entry("Data Federation can be created with private endpoints", Label("datafederation-pe-aws"),
-			model.DataProvider(
-				"privatelink-aws-1",
-				model.NewEmptyAtlasKeyType().UseDefaultFullAccess(),
-				40000,
-				[]func(*model.TestDataProvider){},
-			).WithProject(data.DefaultProject()),
-			[]privateEndpoint{
-				{
-					provider: "AWS",
-					region:   config.AWSRegionEU,
+	It("Creates a data federation with private endpoint", func(ctx context.Context) {
+		var pe *cloud.PrivateEndpointDetails
+		const dataFederationInstanceName = "test-data-federation-aws"
+
+		//nolint:dupl
+		By("Create privaye endpoint in AWS", func() {
+			Expect(testData.K8SClient.Get(testData.Context, types.NamespacedName{Name: testData.Project.Name,
+				Namespace: testData.Resources.Namespace}, testData.Project)).To(Succeed())
+
+			vpcId := providerAction.SetupNetwork(
+				"AWS",
+				cloud.WithAWSConfig(&cloud.AWSConfig{Region: config.AWSRegionEU}),
+			)
+			pe = providerAction.SetupPrivateEndpoint(
+				&cloud.AWSPrivateEndpointRequest{
+					ID:     "vpce-" + vpcId,
+					Region: config.AWSRegionEU,
+					// See https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Data-Federation/operation/createDataFederationPrivateEndpoint
+					ServiceName: "com.amazonaws.vpce.eu-west-2.vpce-svc-052f1840aa0c4f1f9",
```

Comment:
is this just hardcoded on the Atlas (or AWS?) side? is there a risk this ever changes?

Reply:
It's a hard coded value indeed, see the referenced Atlas documentation 🤷

Reply:
nit: Since this depends on the region, I'd make a note that the region can not be changed without changing the service name

Reply:
If this is hard coded, should we make it into a constant?

Reply:
This is an e2e test, I am generally not in favor of constant indirections in the case of tests as they hide wanted assertion values. Having said that, if you insist, I am fine with introducing a constant.

Reply:
If it is just for testing, let's leave it.